lib/main.dart

```diff
@@ -1,33 +1,29 @@
 import 'package:flutter/material.dart';
 import 'package:tank_manager/view/home_page.dart';
 
+import 'package:provider/provider.dart';
+import 'package:tank_manager/model/app_data.dart';
+
 void main() {
   runApp(const MyApp());
 }
 
 class MyApp extends StatelessWidget {
   const MyApp({Key? key}) : super(key: key);
 
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      debugShowCheckedModeBanner: false,
-      title: 'Tank Manager',
-      theme: ThemeData(
-        // This is the theme of your application.
-        //
-        // Try running your application with "flutter run". You'll see the
-        // application has a blue toolbar. Then, without quitting the app, try
-        // changing the primarySwatch below to Colors.green and then invoke
-        // "hot reload" (press "r" in the console where you ran "flutter run",
-        // or simply save your changes to "hot reload" in a Flutter IDE).
-        // Notice that the counter didn't reset back to zero; the application
-        // is not restarted.
-        primarySwatch: Colors.blue,
-        scaffoldBackgroundColor: Colors.grey.shade500,
+    return MultiProvider(
```

Comment:
Do you need `MultiProvider`? See https://docs.flutter.dev/development/data-and-backend/state-mgmt/simple#changenotifierprovider.